### PR TITLE
refactor(config): rename `devModules` to `buildModules`

### PR DESCRIPTION
**Pull request by pi0**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

`devModules`, initially introduced by #5102 to have a way to only include some modules during dev/build time and removing from production dependencies at runtime. It was named after `devDependencies` section in `package.json` as such modules should not be installed for production environment at all. However the original name is proven to be little confusing and making users thinking it is only for development time. New `buildModules` name tries to make it more clear.
However `devModules` was not documented, backward compatibility with deprecation massage is added to support `devModules` until 3.x.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/config/src/options.js [hunk 1]
```diff
@@ -336,6 +336,13 @@ export function getNuxtConfig (_options) {
     consola.warn('build.extractCSS.allChunks has no effect from v2.0.0. Please use build.optimization.splitChunks settings instead.')
   }
 
+  // devModules renamed to buildModules
+  if (typeof options.devModules !== 'undefined') {
+    consola.warn('`devModules` has been renamed to `buildModules`')
```
Comment (manniL) [inline packages/config/src/options.js]: I'd also warn that it's deprecated and will be removed in Nuxt 3.

File: packages/config/src/options.js [hunk 2]
```diff
@@ -336,6 +336,13 @@ export function getNuxtConfig (_options) {
     consola.warn('build.extractCSS.allChunks has no effect from v2.0.0. Please use build.optimization.splitChunks settings instead.')
   }
 
+  // devModules renamed to buildModules
```
Comment (manniL) [inline packages/config/src/options.js]:
```suggestion
  // devModules has been renamed to buildModules. TODO: Remove in Nuxt 3
```

**Review by atinux — DISMISSED** [review]

File: packages/config/src/options.js [hunk 3]
```diff
@@ -336,6 +336,13 @@ export function getNuxtConfig (_options) {
     consola.warn('build.extractCSS.allChunks has no effect from v2.0.0. Please use build.optimization.splitChunks settings instead.')
   }
 
+  // devModules has been renamed to buildModules
+  if (typeof options.devModules !== 'undefined') {
+    consola.warn('`devModules` has been renamed to `buildModules` and will be removed in nuxt 3.')
```
Comment (kevinmarrec) [inline packages/config/src/options.js]:
```suggestion
    consola.warn('`devModules` has been renamed to `buildModules` and will be removed in Nuxt 3.')
```
If it seems relevant

**Review by kevinmarrec — DISMISSED** [review]

**Review by atinux — APPROVED** [review]

**Comment by manniL** [comment]
Great! It's important that we can somehow track the legacy fallbacks we have to remove for Nuxt 3, so adding it in the error message or as a TODO is the best way (to easily `grep` them out) IMO :relaxed:

**Comment by pi0** [comment]
@manniL Good idea. I believe we can do this for all parts in a separate chore PR.

**Pull request merged**